train-winrm.gemspec

```diff
@@ -38,4 +38,10 @@ Gem::Specification.new do |spec|
   spec.add_dependency "chef-winrm", ">= 2.3.11"
   spec.add_dependency "chef-winrm-elevated", ">= 1.2.5"
   spec.add_dependency "chef-winrm-fs", ">= 1.3.7"
+
+  # Gem dependency needed with Ruby 3.4 upgrade
+  spec.add_dependency "syslog", "~> 0.1"
+  spec.add_dependency "csv", "~> 3.0"
+  spec.add_dependency "ostruct", "~> 0.1.0"
+  spec.add_dependency "fiddle", "~> 1.0"
```

Comment:
We could consider adding these gems only for Ruby versions 3.4 and above, since that's when they are needed explicitly.

```ruby
if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("3.4.0")
  spec.add_dependency "syslog", "~> 0.1"
  spec.add_dependency "csv", "~> 3.0"
  spec.add_dependency "ostruct", "~> 0.1.0"
  spec.add_dependency "fiddle", "~> 1.0"
end
```

We could remove this condition once we drop support for Ruby versions below 3.4.

Reply:
Does any condition work in gemspec?

Reply:
I will try this out since it does seem to break in windows for Ruby 3.0 and 3.1. Hopefully it will resolve that. Will test, thanks for the suggestion.